monai/apps/generation/maisi/networks/diffusion_model_unet_maisi.py

```diff
@@ -0,0 +1,415 @@
+                " i.e. `attention_level[i]=False`, the num_head_channels[i] will be ignored."
+            )
+
+        if isinstance(num_res_blocks, int):
+            num_res_blocks = ensure_tuple_rep(num_res_blocks, len(num_channels))
+
+        if len(num_res_blocks) != len(num_channels):
+            raise ValueError(
+                "`num_res_blocks` should be a single integer or a tuple of integers with the same length as "
+                "`num_channels`."
+            )
+
+        if use_flash_attention and not has_xformers:
+            raise ValueError("use_flash_attention is True but xformers is not installed.")
+
+        if use_flash_attention is True and not torch.cuda.is_available():
+            raise ValueError(
+                "torch.cuda.is_available() should be True but is False. Flash attention is only available for GPU."
+            )
+
+        self.in_channels = in_channels
+        self.block_out_channels = num_channels
+        self.out_channels = out_channels
+        self.num_res_blocks = num_res_blocks
+        self.attention_levels = attention_levels
+        self.num_head_channels = num_head_channels
+        self.with_conditioning = with_conditioning
+
+        # input
+        self.conv_in = Convolution(
+            spatial_dims=spatial_dims,
+            in_channels=in_channels,
+            out_channels=num_channels[0],
+            strides=1,
+            kernel_size=3,
+            padding=1,
+            conv_only=True,
+        )
+
+        # time
+        time_embed_dim = num_channels[0] * 4
+        self.time_embed = nn.Sequential(
+            nn.Linear(num_channels[0], time_embed_dim), nn.SiLU(), nn.Linear(time_embed_dim, time_embed_dim)
+        )
+
+        # class embedding
+        self.num_class_embeds = num_class_embeds
+        if num_class_embeds is not None:
+            self.class_embedding = nn.Embedding(num_class_embeds, time_embed_dim)
+
+        self.input_top_region_index = input_top_region_index
+        self.input_bottom_region_index = input_bottom_region_index
+        self.input_spacing = input_spacing
+
+        new_time_embed_dim = time_embed_dim
+        if self.input_top_region_index:
+            self.top_region_index_layer = nn.Sequential(
+                nn.Linear(4, time_embed_dim), nn.SiLU(), nn.Linear(time_embed_dim, time_embed_dim)
+            )
+            new_time_embed_dim += time_embed_dim
+        if self.input_bottom_region_index:
+            self.bottom_region_index_layer = nn.Sequential(
+                nn.Linear(4, time_embed_dim), nn.SiLU(), nn.Linear(time_embed_dim, time_embed_dim)
+            )
+            new_time_embed_dim += time_embed_dim
+        if self.input_spacing:
+            self.spacing_layer = nn.Sequential(
+                nn.Linear(3, time_embed_dim), nn.SiLU(), nn.Linear(time_embed_dim, time_embed_dim)
+            )
+            new_time_embed_dim += time_embed_dim
```

Comment:
Looks like all these layer contains two Linear and SiLu, can we refactor it into a function? Also for `time_embed`.

Reply:
Thanks. I have updated this part. Please proceed.